compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -192,6 +202,60 @@ static LogicalResult setRootConfigForPackPipeline(func::FuncOp entryPointFn,
   return success();
 }
 
+static LogicalResult setRootConfigForPadPackPipeline(func::FuncOp entryPointFn,
+                                                     linalg::MatmulOp matmulOp,
+                                                     AIEConfig cfg) {
+  // ------------------------------------------------------
+  // -------------- Set lowering config -------------------
+  // ------------------------------------------------------
+  // Assume working on a 2x2 AIE array and make sure the tile size is not larger
+  // than the input size.
+  auto initType = matmulOp.getDpsInitOperand(0)->get().getType();
+  auto initShape = llvm::cast<ShapedType>(initType).getShape();
+  auto tileM0 = std::min((int)initShape[0], 64);
+  auto tileN0 = std::min((int)initShape[1], 64);
+  auto tileM1 = std::max((int)tileM0 / 2, 1);
+  auto tileN1 = std::max((int)tileN0 / 2, 1);
+  auto lhsType = matmulOp.getDpsInputOperand(0)->get().getType();
+  auto lhsShape = llvm::cast<ShapedType>(lhsType).getShape();
+  auto tileK0 = std::min((int)lhsShape[1], 256);
+  auto tileK1 = std::min((int)lhsShape[1] / 8, 4);
```

Comment:
There are a bunch of hardcoded values here, which I think might have to do with avoiding for example L2 buffer allocation overflow (`tileM0`, `tileM1`?) and using a 2x2 herd size (`/2`)? Also, I am not sure for example where `tileK0 == 256` exactly comes from? I think it would be useful to document the hardcoded values and any baked-in assumptions in the comments above, for others to understand where these values come from. And long term, I think we should read these values from somewhere else as they might change depending on data type, device, a uKernel to be targeted etc.

Reply:
Tile sizes (64, 64, 256) are chosen empirically for large GEMM sizes. And (tileM1, tileN1) = (tileM0/2, tileN0/2) are chosen to satisfy the 2x2 herd. This is a preliminary setting for tile sizes, we should come up a more sophisticated tiling strategy and a way to choose parameters in the future.